File: app/src/main/java/com/nimtego/plectrum/domain/interactor/AlbumInteractor.java
package com.nimtego.plectrum.domain.interactor;

import com.nimtego.plectrum.presentation.main.model.AlbumModel;

import java.util.List;

import dagger.internal.Preconditions;
import io.reactivex.Observable;

@Deprecated
public class AlbumInteractor
        extends BaseInteractor<List<AlbumModel>, AlbumInteractor.Params> {

    public AlbumInteractor() {
    }

    @Override
    protected Observable<List<AlbumModel>> buildUseCaseObservable(Params params) {
        Preconditions.checkNotNull(params);
        return repository.albums(params.request);
    }

    public static final class Params {

        private final String request;

        private Params(String request) {
            this.request = request;
        }

        public static Params forRequest(String request) {
            return new Params(request);
        }
    }

}

File: app/src/main/java/com/nimtego/plectrum/domain/interactor/AlbumInteractor.kt
package com.nimtego.plectrum.domain.interactor

import com.nimtego.plectrum.data.repository.repository.AlbumRepository
import com.nimtego.plectrum.presentation.mvp.view_model.music.AlbumModel
import io.reactivex.Observable
import io.reactivex.disposables.CompositeDisposable
import javax.inject.Inject


class AlbumInteractor @Inject constructor(
        disposable: CompositeDisposable,
        repository: AlbumRepository
) : BaseInteractor<List<AlbumModel>, AlbumInteractor.Params>(disposable, repository) {
    override fun buildUseCaseObservable(params: Params): Observable<List<AlbumModel>> {
//        Preconditions.checkNotNull(params)
        return repository.query(params.request)
    }

    class Params private constructor(val request: String) {
        companion object {

            fun forRequest(request: String): Params {
                return Params(request)
            }

        }
    }

}